### PR TITLE
Update with react 18next to latest

### LINKED ISSUE #4532: Logic of with-react-i18next is incompatible with _app.js
# Examples bug report / feature request

## Example name

* [with-react-i18next](https://github.com/zeit/next.js/tree/canary/examples/with-react-i18next)
* [with-i18next](https://github.com/zeit/next.js/tree/canary/examples/with-i18next) (possibly)

## Describe the bug  / feature request

Looks like [`with18next()` HOC](https://github.com/zeit/next.js/blob/canary/examples/with-react-i18next/lib/withI18next.js) in `with-react-i18next` example is incompatible with `_app.js` (added in Next v6). Although wrapping each page into the HOC is still possible, this approach has at least these two drawbacks:

* The layout defined in `_app.js` cannot be translated since the components end up _above_ `with18next(['ns'])(PageComponent)`.
* Going between pages flushes the DOM tree since `with18next()` creates separate wrapper components on each page (I may be not 100% here).
 
The `with-apollo` example [has been updated recently](https://github.com/zeit/next.js/pull/4420) and ApolloProvider is now a part of `_app.js`, which is pretty neat. It'd be great if `with-react-i18next` could undergo a similar transformation. I've just tried to propose this change myself, but am giving up after over three hours 😅 

There's one caveat that should be taken into account when implementing the new logic: a collection of namespaces used by a page needs to be passed into `_app.js`. I suggest that these namespaces can be either sourced from `getInitialProps()`:

```jsx
export default class MyPage extends Component {
  static getInitialProps() {
    return { i18nextNamespaces: ['ns1', 'ns2'] };
  }
  render() {
    return <div>{this.props.t('hello')}</div>;
  }
}
```

Alternatively, `i18nextNamespaces` can be defined in a static property to avoid implementing `getInitialProps()` in the simplest cases: 

```jsx
export default class MyPage extends Component {
  static public i18nextNamespaces = ['ns1', 'ns2'];
  render() {
    return <div>{this.props.t('hello')}</div>;
  }
}
```

```jsx
const MyPage = ({t}) => <div>{t('hello')}</div>;
MyPage.i18nextNamespaces = ['ns1', 'ns2'];
export default MyPage;
```

Ideally, these namespaces should be detected automatically depending on which ones are used down the tree (`translate(['ns1'])(Component1)`, `translate(['ns2', 'ns3'])(Component2)`). This magic behaviour would be similar to what Apollo does with executing all queries in the tree on the server and building client-side cache based on that. However, I'm not sure how easy it would be to implement this trick in `react-i18next`.

## System information
 - Version of Next.js: 6.0.3

cc @jamuhl

### PR DESCRIPTION
**Pull request by jamuhl**
Fixes #4532

This updates the with-react-i18next example to latest next.js version with support for the _app.js file.

### CODE REVIEW

**Comment by jamuhl** [comment]
this should also finally close: https://github.com/zeit/next.js/issues/4532

File: examples/with-react-i18next/README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ yarn create next-app --example with-react-i18next with-react-i18next-app
 
 ### Download manually
 
-Download the example:
+Download the example [or clone the repo](https://github.com/zeit/next.js):
```
Comment (timneutkens) [inline examples/with-react-i18next/README.md]: We shouldn't recommend cloning the repo. You're pulling in all of the Next.js source + you have to build it etc.

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Awesome, thanks!

**Comment by jamuhl** [comment]
thank you for all the work done here...

**Pull request merged**